src/main/java/com/github/lwhite1/tablesaw/api/Table.java

```diff
@@ -35,651 +38,677 @@
+     */
+    public String[] columnNameArray() {
+        String[] names = new String[columnList.size()];
+        for (int i = 0; i < columnList.size(); i++) {
+            Column column = columnList.get(i);
+            names[i] = column.name();
+        }
+        return names;
+    }
+
+    @Override
+    public int row(int r) {
+        return r;
+    }
+
+    /**
+     * Returns a string representation of the value at the given row and column indexes
+     *
+     * @param c the column index, 0 based
+     * @param r the row index, 0 based
+     */
+    @Override
+    public String get(int c, int r) {
+        Column column = column(c);
+        return column.getString(r);
+    }
+
+    /**
+     * Returns a table with the same columns as this table, but no data
+     */
+    public Table emptyCopy() {
+        Table copy = new Table(name);
+        for (Column column : columnList) {
+            copy.addColumn(column.emptyCopy());
+        }
+        return copy;
+    }
+
+    /**
+     * Returns a table with the same columns as this table, but no data, initialized to the given row size
+     */
+    public Table emptyCopy(int rowSize) {
+        Table copy = new Table(name);
+        for (Column column : columnList) {
+            copy.addColumn(column.emptyCopy(rowSize));
+        }
+        return copy;
+    }
+
+    /**
+     * Clears all the data from this table
+     */
+    @Override
+    public void clear() {
+        for (Column column : columnList) {
+            column.clear();
+        }
+    }
+
+    /**
+     * Returns a new table containing the first {@code nrows} of data in this table
+     */
+    public Table first(int nRows) {
+        nRows = Math.min(nRows, rowCount());
+        Table newTable = emptyCopy(nRows);
+        Rows.head(nRows, this, newTable);
+        return newTable;
+    }
+
+    /**
+     * Returns a new table containing the last {@code nrows} of data in this table
+     */
+    public Table last(int nRows) {
+        nRows = Math.min(nRows, rowCount());
+        Table newTable = emptyCopy(nRows);
+        Rows.tail(nRows, this, newTable);
+        return newTable;
+    }
+
+    /**
+     * Returns a sort Key that can be used for simple or chained comparator sorting
+     * <p>
+     * You can extend the sort key by using .next() to fill more columns to the sort order
+     */
+    private static Sort first(String columnName, Sort.Order order) {
+        return Sort.on(columnName, order);
+    }
+
+    /**
+     * Returns a copy of this table sorted on the given column names, applied in order,
+     * <p>
+     * if column name starts with - then sort that column descending otherwise sort ascending
+     */
+    public Table sortOn(String... columnNames) {
+        Sort key = null;
+        Order order = null;
+        for (String columnName : columnNames) {
+              order = getOrder(columnName);
+            if (key == null) { // first time through
+                order = getOrder(columnName);
```

Comment:
order = getOrder in the conditional block is redundant. value is set just before the if.  missed this earlier